src/polyswarm_api/types/hunt.py

```diff
@@ -26,6 +30,8 @@ def __init__(self, ruleset, path=None, polyswarm=None):
             raise InvalidYaraRules(*e.args)
 
     def validate(self):
+        if not yara:
+            return True
```

Comment:
Shouldn't we raise an `NotImportedException()` here instead of returning `True`? The user might not be aware that they have to install the package or that the import failed, it that is a possible situation.

The calling function should then capture this exception and log that the package is not imported or break if the validation is enforced (or just stay quiet if there is a quiet flag). What to do in these cases seem to be more of a caller problem than a callee problem.

Reply:
This is intentional behavior. We want to avoid a situation where, in the first time using our product, the user is spammed with a bunch of warnings about something that doesn't actually matter to them. They're told they can get more instructive error messages if they install the package later on.

Reply:
Can't we just have the function that is calling suppressing the log for now (or when verbose logs are not provided)? I don't think it is good that we are returning `True` when the module is not imported. I understand that is the desired behavior, but I don't think this is where it should be treated. I am ok though with this as is, but I think it will need a refactor in the future as this is hiding a exceptional case that the caller method can not act upon.

Reply:
If we raise an exception here, you would never be able to create the object if you don't have `yara-python` installed, which is certainly not the behavior we want. If we move validation out of the constructor, then we can do what you want, which may be the most reasonable course of action.

Reply:
https://github.com/polyswarm/polyswarm-api/pull/72